test/Plutip/Contract.purs

```diff
@@ -784,6 +795,73 @@ suite = do
                 ]
             )
 
+    test "GetScriptByHash" do
+      let
+        distribution :: InitialUTxOs
+        distribution = [ BigInt.fromInt 2_000_000_000 ]
+
+      withWallets distribution \alice -> do
+        withKeyWallet alice do
+          validator1 <- AlwaysSucceeds.alwaysSucceedsScript
+          validator2 <- alwaysSucceedsScriptV2
+          let
+            validatorRef1 = PlutusScriptRef $ unwrap validator1
+            validatorRef2 = PlutusScriptRef $ unwrap validator2
+            useScriptAndGetByHash validator vhash = do
+              txId <- AlwaysSucceeds.payToAlwaysSucceeds vhash
+              awaitTxConfirmed txId
+              -- Spending utxo, to make Kupo (used inside) see the script
+              AlwaysSucceeds.spendFromAlwaysSucceeds vhash validator txId
+              getScriptByHash $ unwrap vhash
+
+          result1 <- useScriptAndGetByHash validator1 (validatorHash validator1)
+          result2 <- useScriptAndGetByHash validator2 (validatorHash validator2)
+
+          -- Testing getScriptByHash
+          result1 `shouldEqual` (Right (Just validatorRef1))
+          result2 `shouldEqual` (Right (Just validatorRef2))
+
+          -- Testing getScriptsByHashes
+          let
+            scriptHash1 = unwrap (validatorHash validator1)
+            scriptHash2 = unwrap (validatorHash validator2)
+          results <- getScriptsByHashes [ scriptHash1, scriptHash2 ]
+          (map unpackResult results) `shouldEqual` Map.fromFoldable
+            [ (scriptHash1 /\ validatorRef1)
+            , (scriptHash2 /\ validatorRef2)
+            ]
+
+    test "GetTxMetadata" do
+      let
+        distribution :: InitialUTxOs
+        distribution = [ BigInt.fromInt 2_000_000_000 ]
+
+      withWallets distribution \alice -> do
+        withKeyWallet alice do
+          tn <- mkTokenName "Token name"
+          mp /\ _ <- mkCurrencySymbol alwaysMintsPolicy
+          let
+            constraints :: Constraints.TxConstraints Void Void
+            constraints =
+              Constraints.mustMintCurrency (mintingPolicyHash mp) tn one
+
+            lookups :: Lookups.ScriptLookups Void
+            lookups =
+              Lookups.mintingPolicy mp
```

Comment:
I think it's sufficient to have `constraints` and `lookups` be `mempty`? I assume you took the following test as a base for the example